src/features/core/hooks/use-is-mounted-ref.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useRef } from 'react';
+
+export function useIsMountedRef() {
+  const ref = useRef({ isMounted: true }).current;
+
+  useEffect(() => {
+    ref.isMounted = true;
+    return () => {
+      ref.isMounted = false;
+    };
+  }, [ref]);
+
+  return ref;
```

Comment:
```suggestion
  const isMounted = useRef(true);

  useEffect(() => {
    isMounted.value = true;
    return () => {
      isMounted.value = false;
    };
  }, [ref]);

  const getIsMounted = () => isMounted.value;
  
  return { getIsMounted };
```

ref: https://stackoverflow.com/a/74728579